### PR TITLE
Rework zip code fetching and import CCVI data via dynamic import / hooks.

### PR DESCRIPTION
**Pull request by mikelehen**
This moves us in the direction of using dynamic `import()` for our JSON data loading which should have a significant caching advantage since our chunk files are cached forever (which is safe since the filename is based on a hash of the content).

* Added a `importJson()` helper for dynamically importing JSON chunks.
* Added a `usePromise()` helper that wraps a Promise into a React hook.
* Added `useZipCodeToFipsMap()` and `useFipsToCcviMap()` hooks to demonstrate how to use the helpers.
* Renamed ccvi-data.json to ccvi.json
* Moved county-zipcode.json from data/ to src/ so that it gets compiled by webpack and can be dynamically imported as a chunk.

### CODE REVIEW

File: src/common/hooks/useCountyToZipMap.ts [hunk 1]
```diff
@@ -1,51 +1,8 @@
...
-  return {
-    countyToZipMap: zipData,
-    isLoading,
-  };
-};
-
-const useCountyToZipMap = singletonHook(init, useCountyToZipMapImpl);
+const useCountyToZipMap = singletonHook({ pending: true }, () =>
+  usePromise(importCountyToZipCodeMap()),
+);
```
Comment (mikelehen) [inline src/common/hooks/useCountyToZipMap.ts:7]: FWIW- I tried a bit to create a helper to abstract away this boilerplate (using `singletonHook` and `usePromise` together), but had some trouble and I am in meetings most of the day and wanted to get this out.  I might try again later, though when monkeying with this stuff I kept tripping over https://reactjs.org/docs/hooks-rules.html

File: src/common/hooks/usePromise.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * Helper to wrap an asynchronous promise result into a React Hook. The current state
+ * of the Promise will be returned each time the hook is called as a `PromiseState`.
+ *
+ * @param promise The promise to wrap. Must not change between calls or the hook will re-run.
+ * @returns The current state of the promise.
+ */
+export default function usePromise<T>(promise: Promise<T>): PromiseState<T> {
+  const [result, setResult] = useState<PromiseState<T>>({ pending: true });
+  useEffect(() => {
+    let canceled = false;
```
Comment (chasulin) [inline src/common/hooks/usePromise.ts:25]: whats the purpose of `canceled`

File: src/common/data/index.ts [hunk 3]
```diff
@@ -17,8 +15,43 @@ export interface RegionCcviItem {
...
+  importPromise: Promise<{ default: T }>,
+): Promise<T> {
+  cachedImports[cacheToken] =
+    cachedImports[cacheToken] || importPromise.then(module => module.default);
+  return cachedImports[cacheToken];
+}
+
+/** Dynamic import for ccvi.json. */
+export function importFipsToCcviMap(): Promise<FipsToCcviMap> {
+  return importJson('ccvi', import('./ccvi.json'));
```
[the remark this answers is not in the document]
Comment (pnavarrc) [reply src/common/data/index.ts:51]: I recently learned that `import` is not actually a function is just syntax that looks like a function

**Review by pnavarrc — APPROVED** [review]
This is awesome, thanks for the detailed explanation!

File: src/common/data/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (schamp) [reply src/common/data/index.ts:51]: nice approach!

[the remark this answers is not in the document]
Comment (mikelehen) [reply src/common/data/index.ts:51]: Ah, interesting! I guess that makes sense.  It definitely behaves a bit magically. 😁

File: src/common/hooks/usePromise.ts (hunk 2 above)
Comment (chasulin) [reply src/common/hooks/usePromise.ts:25]: ahh i see

**Pull request merged**